### PR TITLE
Fix: The values of StatusCodes shown on Swagger UI of mentorship_relation API changed  to be in number format instead of Enum Representations.

### LINKED ISSUE #939: Bug:Use HTTPStatus code values instead  of enum representation, in mentorship_relation API in Swagger UI
**Describe the bug**

The values shown on Swagger UI of mentorship_relation API should be in number format instead of Enum Representations of Status Codes. For example 200 instead of HTTPStatus.OK. 

**To Reproduce**

1. Go to https://mentorship-backend-temp.herokuapp.com/
2. Scroll down to ```mentorship_relation```
3. See enum representation.
4. 
image

**Expected behavior**
Use of Values of Status Codes instead of Enum Representations in Swagger UI

**Desktop (please complete the following information):**
 - OS: Ubuntu 20+
 - Browser FireFox Web Browser
 - Version --

### PR DESCRIPTION
**Pull request by Aaishpra**
### Description
The values of  HTTPStatus codes shown on Swagger UI of mentorship_relation API changed to be in number format instead of Enum Representations. 

Fixes #939

### Type of Change:

- Code

- User Interface

**Code/Quality Assurance Only**
- Bug fix (non-breaking change which fixes an issue)

### How Has This Been Tested?
Go to ``` mentorship_relation```
Click on ```Post/mentorship_relation/send_request```(or any other task)
See Results

image

### Checklist:

- [x] My PR follows the style guidelines of this project
- [x] I have performed a self-review of my own code or materials

**Code/Quality Assurance Only**

- [x] My changes generate no new warnings 
- [x] My PR currently breaks something (fix or feature that would cause existing functionality to not work as expected)
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been published in downstream modules

### CODE REVIEW

**Comment by Aaishpra** [comment]
@isabelcosta @devkapilbansal can anyone of you review it?
Also, I don't understand why these tests are failing.

**Comment by devkapilbansal** [comment]
@Aaishpra these tests failed due to some dependencies issues which will soon be resolved

**Comment by isabelcosta** [comment]
@Aaishpra can you please update your branch to have the latest changes? These will fix the tests :) hopefully 🙏 
cc @devkapilbansal

**Comment by Aaishpra** [comment]
> @Aaishpra can you please update your branch to have the latest changes? These will fix the tests :) hopefully pray
> cc @devkapilbansal

Done, but still some tests are failing.

**Comment by isabelcosta** [comment]
@Aaishpra the tests are passing fine. don't worry about **codecov/patch** check, this is a common fail, we are not sure why yet. Regarding code formatting, if you run `black app/api/resources/mentorship_relation.py` this will reformat code according to [black](https://github.com/psf/black) tool standards

**Comment by Aaishpra** [comment]
> @Aaishpra the tests are passing fine. don't worry about **codecov/patch** check, this is a common fail, we are not sure why yet. Regarding code formatting, if you run `black app/api/resources/mentorship_relation.py` this will reformat code according to [black](https://github.com/psf/black) tool standards

Ahh thanks, for telling.
Another thing can i get assigned to any one of the new issues which i opened😁
#946 #947

**Comment by isabelcosta** [comment]
> > @Aaishpra the tests are passing fine. don't worry about **codecov/patch** check, this is a common fail, we are not sure why yet. Regarding code formatting, if you run `black app/api/resources/mentorship_relation.py` this will reformat code according to [black](https://github.com/psf/black) tool standards
> 
> Ahh thanks, for telling.
> Another thing can i get assigned to any one of the new issues which I opened😁

I would prefer if you could fix this issue first, then we can leave another available, and once you finished your change requests here you can move on to the next one, and leave another one for a first-time contributor. I will label them accordingly. So if by the time you finish your part in this PR, you can move to another one. Just ask there to be assigned to it :) @Aaishpra

**Comment by Aaishpra** [comment]
> > > @Aaishpra the tests are passing fine. don't worry about **codecov/patch** check, this is a common fail, we are not sure why yet. Regarding code formatting, if you run `black app/api/resources/mentorship_relation.py` this will reformat code according to [black](https://github.com/psf/black) tool standards
> > 
> > Ahh thanks, for telling.
> > Another thing can i get assigned to any one of the new issues which I opened😁
> 
> I would prefer if you could fix this issue first, then we can leave another available, and once you finished your change requests here you can move on to the next one, and leave another one for a first-time contributor. I will label them accordingly. So if by the time you finish your part in this PR, you can move to another one. Just ask there to be assigned to it :) @Aaishpra 

Sure thing 👍

**Comment by devkapilbansal** [comment]
> @Aaishpra the tests are passing fine. don't worry about **codecov/patch** check, this is a common fail, we are not sure why yet. Regarding code formatting, if you run `black app/api/resources/mentorship_relation.py` this will reformat code according to [black](https://github.com/psf/black) tool standards

@isabelcosta may be I know the reason why codecov/patch failed.

https://github.com/anitab-org/mentorship-backend/blob/9618e2c697f5f4c0bdc851a887e4a2a47a39122a/tests/mentorship_relation/test_api_send_request.py#L89

Here, hard coded status code is used, probable fixing these in tests can increase codecov/patch coverage.
Also, most of the tests are not covered in the tests file, like every possible error should be there in the test. We can work on that too

**Comment by Aaishpra** [comment]
> @Aaishpra the tests are passing fine. don't worry about **codecov/patch** check, this is a common fail, we are not sure why yet. Regarding code formatting, if you run `black app/api/resources/mentorship_relation.py` this will reformat code according to [black](https://github.com/psf/black) tool standards

@isabelcosta Done, please review and suggest any other change if required.

File: app/api/resources/mentorship_relation.py [hunk 1]
```diff
@@ -92,12 +93,12 @@ def post(cls):
         is_valid = SendRequest.is_valid_data(data)
 
         if is_valid != {}:
-            return is_valid, HTTPStatus.BAD_REQUEST
+            return is_valid, HTTPStatus.BAD_REQUEST.value
```
Comment (devkapilbansal) [inline app/api/resources/mentorship_relation.py]:
```suggestion
            return is_valid, HTTPStatus.BAD_REQUEST
```

**Review by devkapilbansal — CHANGES_REQUESTED** [review]
@Aaishpra remove `.value` from return statements as asked by @isabelcosta [here](https://github.com/anitab-org/mentorship-backend/pull/952#discussion_r549875920)

File: app/api/resources/mentorship_relation.py [hunk 2]
```diff
@@ -92,12 +93,12 @@ def post(cls):
...
 
         response = DAO.create_mentorship_relation(user_sender_id, data)
 
         # if the mentorship relation creation failed dont send email and return
-        if response[1] != HTTPStatus.CREATED:
+        if response[1] != HTTPStatus.CREATED.value:
```
Comment (Aaishpra) [inline app/api/resources/mentorship_relation.py:101]: shall I remove .value from here too.
@devkapilbansal

Comment (devkapilbansal) [reply app/api/resources/mentorship_relation.py:101]: No, it looks fine

Comment (Aaishpra) [reply app/api/resources/mentorship_relation.py:101]: Done @devkapilbansal

**Comment by Aaishpra** [comment]
@devkapilbansal @isabelcosta  please review the changes.

**Review by vj-codes — CHANGES_REQUESTED** [review]
@Aaishpra can you please remove the change made in` requirements.txt` as it does not deal with the issue, thanks!

**Comment by Aaishpra** [comment]
> @Aaishpra can you please remove the change made in` requirements.txt` as it does not deal with the issue, thanks!

I had to update my fork, because of some failing check that's why there is a change reflected in 'requirement.txt'.

**Comment by devkapilbansal** [comment]
> > @Aaishpra can you please remove the change made in` requirements.txt` as it does not deal with the issue, thanks!
> 
> I had to update my fork, because of some failing check that's why there is a change reflected in 'requirement.txt'.

@Aaishpra that change in requirements may be there because tip of your branch is behind. You should update it with latest develop branch.

**Comment by Aaishpra** [comment]
@devkapilbansal I don't think the change in requirement.txt will cause any problem after this PR is merged as the change is already there in the develop branch of this repo, but I think updating my fork again will cause only extra commits in the PR which ain't required. 
If you still think updating my fork is required just ping me again I will update it.

**Comment by devkapilbansal** [comment]
> @devkapilbansal I don't think the change in requirement.txt will cause any problem after this PR is merged as the change is already there in the develop branch of this repo, but I think updating my fork again will cause only extra commits in the PR which ain't required.
> If you still think updating my fork is required just ping me again I will update it.

It may not cause extra commits. You can soft reset your changes to current develop branch.
Something like:- 
```
git remote add upstream https://github.com/anitab-org/mentorship-backend.git
git upstream fetch develop
git reset --soft upstream/develop
git commit -m "Your message here"
```
Hope it helps 😊

**Comment by Aaishpra** [comment]
@devkapilbansal thanks for the help although I think messed somethings which have caused all this mess.

That's why closing this PR.

**Pull request closed without merging**